src/storage/txn/process.rs

```diff
@@ -600,8 +601,9 @@ fn process_write_impl<S: Snapshot, L: LockMgr>(
             } else {
                 let lock = lock_manager::extract_lock_from_result(&locks[0]);
```

Comment:
Could we check if `timeout < 0` here so that we can avoid generate useless key hashes?

Reply:
Lock manager is a trait and storage shouldn't know how to handle wait-for-lock. I think the overhead here is trivial.

Reply:
If we make `extract_lock_from_result` a trait mehod and pass option in, the overhead can be avoided.

Anyway, this is not so important.